src/shared/Core/Interop/MacOS/MacOSEnvironment.cs

```diff
@@ -0,0 +1,34 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+using GitCredentialManager.Interop.Posix;
+
+namespace GitCredentialManager.Interop.MacOS
+{
+    public class MacOSEnvironment : PosixEnvironment
+    {
+        public MacOSEnvironment(IFileSystem fileSystem)
+            : this(fileSystem, GetCurrentVariables()) { }
+
+        internal MacOSEnvironment(IFileSystem fileSystem, IReadOnlyDictionary<string, string> variables)
+            : base(fileSystem)
+        {
+            EnsureArgument.NotNull(variables, nameof(variables));
+            Variables = variables;
+        }
```

Comment:
We don't need to replicate the constructor of `PosixEnvironment` here – just call through to the corresponding `base` constructor.

```suggestion
        public MacOSEnvironment(IFileSystem fileSystem)
            : base(fileSystem) { }

        internal MacOSEnvironment(IFileSystem fileSystem, IReadOnlyDictionary<string, string> variables)
            : base(fileSystem, variables) { }
```